graylog2-web-interface/src/components/configurations/CertificateRenewalPolicyConfig.tsx

```diff
@@ -0,0 +1,258 @@
+                      bsSize="xs"
+                      onClick={() => {
+                        setShowModal(true);
+                      }}>Edit configuration
+              </Button>
+            </IfPermitted>
+          </p>
+        </>
+      )}
+
+      <Modal show={showModal}
+             onHide={resetConfig}
+             aria-modal="true"
+             aria-labelledby="dialog_label"
+             data-app-section="configurations_index_defaults"
+             data-event-element={modalTitle}>
+        <Formik<FormConfig> onSubmit={saveConfig} initialValues={formConfig}>
+          {({ values, setFieldValue, isSubmitting }) => (
+            <Form>
+              <Modal.Header closeButton>
+                <Modal.Title id="dialog_label">{modalTitle}</Modal.Title>
+              </Modal.Header>
+
+              <Modal.Body>
+                <div>
+                  <Row>
+                    <Col md={12}>
+                      <Field name="mode">
+                        {({ field: { name, value, onChange } }) => (
+                          <Input id={name}
+                                 label="Certificate Renewal Mode"
+                                 help={renewalModeExplanation}>
+                            <Select options={certicateRenewalModes}
+                                    clearable={false}
+                                    name={name}
+                                    value={value ?? 'AUTOMATIC'}
+                                    aria-label="Select certificate renewal mode"
+                                    size="small"
+                                    onChange={(newValue) => onChange({ target: { name, value: newValue } })} />
+                          </Input>
+                        )}
+                      </Field>
+                      <TimeUnitInput label="Certificate Lifetime"
+                                     help={lifetimeExplanation}
+                                     update={(value, unit) => {
+                                       setFieldValue('lifetimeValue', value);
+                                       setFieldValue('lifetimeUnit', unit);
+                                     }}
+                                     value={values.lifetimeValue}
+                                     unit={values.lifetimeUnit.toLocaleUpperCase()}
+                                     enabled
+                                     hideCheckbox
+                                     units={TIME_UNITS_UPPER} />
+                    </Col>
+                  </Row>
+                </div>
+              </Modal.Body>
+
+              <Modal.Footer>
+                <Button type="button" onClick={resetConfig}>Cancel</Button>
+                <Button type="submit"
+                        bsStyle="success"
+                        disabled={isSubmitting}>{isSubmitting ? 'Updating configuration' : 'Update configuration'}
+                </Button>
+              </Modal.Footer>
```

Comment:
Here we can use the `ModalSubmit` component.

Reply:
:white_check_mark: